pkg/util/azureclient/applens/applens_client.go

```diff
@@ -61,16 +61,20 @@ func NewClient(endpoint, issuerUrlTemplate, caName, scope string, cred azcore.To
 }
 
 func newPipeline(authPolicy []policy.Policy, options *ClientOptions, issuerUrlTemplate, caName string) (*runtime.Pipeline, error) {
-	var cp *x509.CertPool = nil
-	var err error = nil
 	if options == nil {
+		var cp *x509.CertPool
 		// if provided pki info fetch the correct cert pool
 		// otherwise use the default of nil
 		if issuerUrlTemplate != "" && caName != "" {
-			cp, err = pki.GetTlsCertPool(issuerUrlTemplate, caName)
+			data, err := pki.FetchDataFromGetIssuerPki(issuerUrlTemplate, caName)
 			if err != nil {
 				return nil, err
 			}
+
+			cp, err := pki.BuildCertPoolForCaName(data)
+			if err != nil || cp == nil {
```

Comment:
I think the _cp == nil_ part is redundant and can be deleted.

Reply:
In the current implementation this is definitely true but more generally I would expect a builder that is unable to build -> invalid state so it should never give a `nil, nil`.

Reply:
Changed the approach to just return the NewClientOptions from inside the function as we just talked about 🙂